src/HexagonOptimize.cpp

```diff
@@ -246,6 +246,11 @@ Expr replace_pattern(Expr x, const vector<Expr> &matches, const Pattern &p) {
     return x;
 }
 
+bool is_double_vector(Expr x, const Target &target) {
+    int native_vector_bits = target.natural_vector_size(Int(8)) * 8;
+    return (x.type().lanes() / ( native_vector_bits / x.type().bits()) >= 2);
```

Comment:
I think this needs to be % 2 == 0, not >= 2? Or more clearly (IMO):

    int native_vector_lanes = target.natural_vector_size(x.type());
    return x.type.lanes() % (2 * native_vector_lanes) == 0;

Reply:
Changed. Thanks.